f3probe.c

```diff
@@ -474,7 +483,7 @@ static int test_device(struct args *args)
 	case FKTY_LIMBO:
 	case FKTY_WRAPAROUND:
 	case FKTY_CHAIN: {
-		uint64_t last_good_sector = (real_size_byte >> 9) - 1;
+		last_good_sector = (real_size_byte >> 9) - 1;
 		assert(block_order >= 9);
 		printf("Bad news: The device `%s' is a counterfeit of type %s\n\n"
```

Comment:
1. Before the `printf()`, add the following line:
```
assert(asprintf(&fix_cmd, "f3fix --last-sec=%" PRIu64 " %s", last_good_sector, final_dev_filename) > 0);
```
Break into multiple lines as needed.

2. Edit the `printf()` to use variable `fix_cmd` instead of repeating it.